frontend/dispatcher.go

```diff
@@ -17,6 +17,47 @@ type HTTPError struct {
 	Message string
 }
 
+func db_requester() {
+	for {
+		db_request := <-db_open_channel
```

Comment:
Use `select` instead and add the ability to shutdown the goroutine on sigterm or other signals.
 
Also, lets avoid this global channel and maybe use a function that spawns a goroutine and _returns_ a channel instead:

```go
func db_requester() chan DBOpenRequest {
    c := make(chan DBOpenRequest)

}
```